test/pylib/resource_gather.py

```diff
@@ -54,6 +58,12 @@ def get_cgroup() -> Path:
 
 class ResourceGather(ABC):
     def __init__(self, test: TestPyTest):
+        try:
+            self.loop = asyncio.get_running_loop()
+            self.own_loop = False
+        except RuntimeError:
+            self.loop = asyncio.new_event_loop()
+            self.own_loop = True
```

Comment:
Please, explain this change. I don't see how it fits the patch detailed description (despite it definitely fits the generic "refactor" title :D )

Reply:
looks like it is change(preparation) for pytest multithreading(xdist) support. Now in Master, we already support running boost tests via pure pytest command, but without ResourceGather support. In CI, ResourceGather is enabled by default, so we need to support it  to run boost tests via pytest in CI

Reply:
Modified commit message and added a comment